docroot/modules/humsci/hs_layouts/hs_layouts.install

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/**
+ * @file
+ * Update functions for hs_layouts module.
+ */
+
+use Drupal\layout_builder\Section;
+
+/**
+ * Set main content id in content types default layout builder config.
+ */
+function hs_layouts_update_9401() {
+  $storage = \Drupal::entityTypeManager()->getStorage('entity_view_display');
+  $query = $storage->getQuery()
+    ->condition('targetEntityType', 'node')
+    ->condition('mode', 'default');
+  $items = $query->execute();
+  foreach ($items as $item) {
+    $view_display = $storage->load($item);
+    if ($view_display->isLayoutBuilderEnabled()) {
+      $sections = $view_display->getThirdPartySetting('layout_builder', 'sections');
+      if (count($sections)) {
+        $main_section = $sections[0];
+        if (count($sections) > 1) {
+          foreach ($sections as $section) {
+            if ($section->getLayoutSettings()['section_width'] !== 'hs-full-width') {
+              $main_section = $section;
+              break;
+            }
+          }
+        }
+        $main_section->setLayoutSettings(array_merge($main_section->getLayoutSettings(), ['main_content' => 'main-region']));
+        $view_display->save();
+      }
+    }
+  }
+}
+
+/**
+ * Set main content id in nodes with overridden layout builder config.
+ */
+function hs_layouts_update_9402() {
+  // Get all sections in nodes with overridden layout.
+  $db = \Drupal::database();
+  $result = $db->select('node__layout_builder__layout', 'nl')
+    ->fields('nl', ['entity_id', 'delta', 'layout_builder__layout_section'])
+    ->orderBy('entity_id')
+    ->orderBy('delta')
+    ->execute();
```

Comment:
use entity query and refactor the upgrade process. Do the work on the node and sections, not the database record. This is far too dangerous and complicated. Do basically the same as the previous update hook.

Reply:
Fixed. However, there's a problem when running the update hooks in some specific sites. We're getting the following errors:

Reply:
@cienvaras Is there a specific site you're experiencing this on? I'd like to test it on my local. I'm also not sure if this would actually be an issue on Acquia.

Reply:
@joegl Yes, in sites that have an underscore in their name: `hs_traditional`, `hs_colorful`, `culture_emotion_lab`, etc.

I did some tests in the acquia dev environment last year and the problem was present, because drush used the `http://[SITENAME]` as the domain in the CI. But I'm not sure if the problem is still present.

Reply:
Oh right, those would be good examples to try based on your first report 🤦